ykcompile/src/asm_helpers.rs

```diff
@@ -0,0 +1,30 @@
+//! Helpers for generating assembler code with dynasm.
+
+/// Emits a 'mem <- reg'  assembler instruction using the desired size qualifier.
+macro_rules! asm_mem_reg {
```

Comment:
We could merge `asm_mem_reg` and `asm_reg_mem` into one, by parsing the expressions. Though I am not sure if it's worth it.

Reply:
Also, will we be needing a `asm_mem_mem` at some point?

Reply:
Is that even encodable in x86_64?

Reply:
You are right, it's not. And it's probably best to leave the spilling outside of the helper.

Reply:
>  probably best to leave the spilling outside of the helper.

Why's that?

Reply:
More for aesthetic reasons as we'd have to forward the register allocator to the helper function.